src/main/java/com/celements/web/plugin/cmd/ExternalJavaScriptFilesCommand.java

```diff
@@ -35,127 +40,224 @@
+import com.celements.javascript.ExtJsFileParameter;
+import com.celements.javascript.JsLoadMode;
 import com.celements.model.access.IModelAccessFacade;
 import com.celements.model.access.exception.DocumentNotExistsException;
-import com.celements.pagetype.PageTypeReference;
+import com.celements.model.context.ModelContext;
+import com.celements.model.object.xwiki.XWikiObjectFetcher;
+import com.celements.model.reference.RefBuilder;
 import com.celements.pagetype.service.IPageTypeResolverRole;
 import com.celements.pagetype.xobject.XObjectPageTypeUtilsRole;
-import com.celements.sajson.Builder;
-import com.celements.web.classcollections.OldCoreClasses;
+import com.celements.sajson.JsonBuilder;
+import com.celements.web.classcollections.IOldCoreClassConfig;
 import com.celements.web.service.IWebUtilsService;
+import com.google.common.base.Strings;
+import com.google.errorprone.annotations.Immutable;
 import com.xpn.xwiki.XWikiContext;
-import com.xpn.xwiki.XWikiException;
 import com.xpn.xwiki.api.Document;
 import com.xpn.xwiki.doc.XWikiDocument;
-import com.xpn.xwiki.objects.BaseObject;
 import com.xpn.xwiki.web.Utils;
 
+import one.util.streamex.StreamEx;
+
 public class ExternalJavaScriptFilesCommand {
 
+  /**
+   * @deprecated since 5.4 instead use IOldCoreClassConfig.JAVA_SCRIPTS_EXTERNAL_FILES_CLASS_DOC
+   */
+  @Deprecated
   public static final String JAVA_SCRIPT_EXTERNAL_FILES_CLASS_DOC = "ExternalFiles";
+  /**
+   * @deprecated since 4.0 instead use IOldCoreClassConfig.JAVA_SCRIPTS_EXTERNAL_FILES_CLASS_SPACE
+   */
+  @Deprecated
   public static final String JAVA_SCRIPT_EXTERNAL_FILES_CLASS_SPACE = "JavaScript";
+  /**
+   * @deprecated since 4.0 instead use IOldCoreClassConfig.JAVA_SCRIPTS_EXTERNAL_FILES_CLASS
+   */
+  @Deprecated
   public static final String JAVA_SCRIPT_EXTERNAL_FILES_CLASS = JAVA_SCRIPT_EXTERNAL_FILES_CLASS_SPACE
       + "." + JAVA_SCRIPT_EXTERNAL_FILES_CLASS_DOC;
 
   private static final Logger LOGGER = LoggerFactory.getLogger(
       ExternalJavaScriptFilesCommand.class);
 
-  private XWikiContext context;
-  private Set<String> extJSfileSet;
-  private Set<String> extJSAttUrlSet;
-  private List<String> extJSfileList;
-  private List<String> extJSnotFoundList;
-  private boolean displayedAll = false;
-  private AttachmentURLCommand attUrlCmd_injected = null;
-  private IPageTypeResolverRole ptResolver_injected = null;
+  @Immutable
+  static final class JsFileEntry {
+
+    private final String jsFileUrl;
+    private final JsLoadMode loadMode;
+
+    JsFileEntry(String jsFileUrl) {
+      this(jsFileUrl, (JsLoadMode) null);
+    }
+
+    JsFileEntry(String jsFileUrl, @Nullable JsLoadMode loadMode) {
+      this.jsFileUrl = jsFileUrl;
+      this.loadMode = Optional.ofNullable(loadMode).orElse(JsLoadMode.SYNC);
+    }
+
+    JsFileEntry(String jsFileUrl, @Nullable String loadModeStr) {
+      this(jsFileUrl, JsLoadMode.convertStoreValue(loadModeStr));
+    }
```

Comment:
I don't think this constructor is useful, the client can convert the enum itself